### DISCUSSION TITLE
Use of time_budget to limit execution time

### DISCUSSION THREAD

**Discussion by Marcovnr** [winedarksea/AutoTS#149] [Q&A]
Hello guys! How are you all doing?

I am using AutoTS frequently now, but I have been struggling in setting a limit to the execution time while fitting the AutoTS.

I wanted to know if there is such a feature already implemented, or if there are any perspective of having this feature. I do know I can use the keyboard to interrupt the execution, however I would be interested in setting the limit by actual time.

Thanks and hope to hear from any of you soon :)

**Comment by winedarksea** [comment]
There are a couple of options for model_interrupt which allows you to skip just one model when =True or 'end_generation' to end the entire generation.

I have always wanted to add a timeout just like you described, but the problem is I would have to check the start time vs run time, and I can't do that inside the model, only before/after each model runs. So it doesn't help at all for situations where just a few models are being unreasonably slow, which is the more common problem. It would only help where you want to vaguely limit the end time, not an exact end time. IDK if that would be helpful?

**Reply by Marcovnr** [reply]
That would be very helpful! The problem is basically having no idea of when you should expect the process to finish. If it were possible for at least having an approximate time limit, that would be great!

**Reply by Marcovnr** [reply]
For instance, let´s say I want to use a model in the multivariate list. Ideally, I would just run it overnight and that´s it. However sometimes I have to make a quick training to test another part of my code, sacrificing other models to test. It would be great to have a parameter where I can just say for the AutoTS that I am happy with the best model you find in the first 20 minutes (more or less).

**Reply by winedarksea** [reply]
I just added `generation_timeout` to the dev branch. Probably won't be on a pypi release for a few weeks. With this, it will end the search space after _n_ minutes (checking after each generation ends, only) and then after this timeout, will then proceed to validation (validation then taking possibly the same amount of time again, depending on validation params).
